### ISSUE TITLE
NCK tests fail intermittently

### ISSUE THREAD

**Issue by billsacks** [CESM-Development/cime#116]
I have noticed that NCK tests fail intermittently when they are run as part of a test suite.

In particular: The second run (which is supposed to be a two-instance run) runs with only a single instance. Oddly, though, its output files have suffixes of _0001 - so it is in some hybrid state between a 1-instance run and a 2-instance run.

This is hard to reproduce consistently, which makes me suspect a race condition. Specifically, there could be a race condition with respect to sharedlibroot, where the NCK test builds, then another test builds before NCK executes, and that second test's build stomps on the sharedlib that NCK needs.

From spot-checking a few cases of success and failure, I saw some evidence that this could be happening... these all refer to NCK_Ld1.f10_f10.ICRUCLM45.yellowstone_intel.clm-default:

(1) FAILed NCK: Build done between the NCK build and run was ERP_P15x2_Lm13.f10_f10.IHISTCLM45BGC.yellowstone_intel.clm-monthly, which I think used the same sharedlibroot (other than the number of instances)

(2) PASSed NCK: Build done between the NCK build and run was SSP_D_Ld10.f19_g16.I1850CLM45BGC.yellowstone_intel.clm-default, which used a different sharedlibroot (debug rather than nodebug)

(3) PASSed NCK: Build done between the NCK build and run was ERS_Ld5.1x1_vancouverCAN.I1PTCLM50.yellowstone_intel.clm-default, which used a different sharedlibroot (mpi-serial rather than mpich2)

_However:_

(4) FAILed NCK: Build done between the NCK build and run was ERP_Ln9.f19_f19.FW5.yellowstone_intel.cam-outfrq9s (csm_share built before the NCK ran; clm did not)... this used a different sharedlibroot (threads rather than nothreads), so this fails to support my hypothesis.

**Comment by billsacks** [comment]
cc'ing @jedwards4b @sholly @mvertens @ekluzek @bandre-ucar

**Comment by billsacks** [comment]
The ideal resolution in my mind would be to make the number of instances a runtime parameter rather than compile-time. However, I don't know if that's feasible.

**Comment by billsacks** [comment]
The specific failure for the NCK test is:

```
FAIL NCK_Ld1.f10_f10.ICRUCLM45.yellowstone_intel.clm-default.clm2.h0.nc : test compare clm2.h0 (.base and .multiinst for _0002) (no history file in test case)
FAIL NCK_Ld1.f10_f10.ICRUCLM45.yellowstone_intel.clm-default.clm2.h1.nc : test compare clm2.h1 (.base and .multiinst for _0002) (no history file in test case)
FAIL NCK_Ld1.f10_f10.ICRUCLM45.yellowstone_intel.clm-default : test functionality summary (compare .base with .multiinst) (no history file in test case)
```

**Comment by jedwards4b** [comment]
I have not been able to reproduce this failure, closing this item.

**Comment by billsacks** [comment]
From the CSEG meeting today, it sounds like this issue remains for multi-instance and ESMF tests. So I am reopening it, partly so we can keep track of it for the sake of CLM testing.

**Comment by bandre-ucar** [comment]
It sounds like the same problem occurs in clm4_5_1_r120, using cime2.0.7. 

Without diving into the MCT, it looks like a compile time and runitme arg don't agree?

  12:MCT::m_Rearranger::Rearrange_: SourceAV size is not appropriate for this Re
arranger
  12:MCT::m_Rearranger::Rearrange_: error, InRearranger%SendRouter%lAvsize=6695,
 AttrVect_lsize(SourceAV)=0.
  12:00C.MCT(MPEU)::die.: from MCT::m_Rearranger::Rearrange_()
  12:Abort(2) on node 12 (rank 12 in comm 1140850688): application called MPI_Abort(MPI_COMM_WORLD, 2) - process 12

**Comment by ekluzek** [comment]
I had this same problem with clm4_5_1_r119. The tests worked, after cleaning the build, rebuilding and resubmitting.

**Comment by billsacks** [comment]
This appears to be fixed (by #148 ?); closing it.
